### DISCUSSION TITLE
0.23.0-rc2: auxiliary.db not created with correct tables when existing data.db is present or after auxiliary.db deletion

### DISCUSSION THREAD

**Discussion by lumosminima** [pocketbase/pocketbase#5618] [Q&A] [answered]
If I have existing `pb_data` directory with only a single `data.db` which is migrated to `0.23.0-rc2` then doing `./pocketbase serve --dev` I get:

![Screenshot 2024-10-07 160907](https://github.com/user-attachments/assets/a9eff64e-8ffc-469c-b8a4-ff44fa5eb68e)

This was with my existing `pb_data` so I tried with the binary from releases and it creates the proper auxiliary.db with `_logs` table only if there isn't already a `data.db` file present. So, if you delete the `auxiliary.db` after creation it doesn't seem to migrate properly. For now, I'm just copying the correctly created auxiliary.db which fixes the logs for my app.

**OS:** WSL2 (Ubuntu 24.04.1 LTS)
**GO:** go version go1.23.2 linux/amd64
**PB**: 0.23.0-rc2

**Comment by ganigeorgiev** [comment, marked as answer]
With v0.23 the migration history is kept only in 1 place (`pb_data/data.db`) to prevent inconsistencies but this unfortunately can cause errors like the above when deleting the `auxiliary.db` file because the `init.go` system migration technically has been already executed and its state is already stored in the `_migrations` table so it doesn't run second time.

I'm not sure what would be the best option here because in the future the `auxiliary.db` is intended to store other things beside logs (aka. it may not be ephemeral) but I've added in my todo to consider with the final v0.23.0 release to maybe split the aux migrations and/or rerun only the minimal aux migrations in case the db is empty.

**Reply by ganigeorgiev** [reply]
_To further clarify, the above error could happen only if you manually delete the `auxiliary.db` file after the application has been already started/migrated because [`1640988000_init.go`](https://github.com/pocketbase/pocketbase/blob/develop/migrations/1640988000_init.go) and [`1717233556_v0.23_migrate.go`](https://github.com/pocketbase/pocketbase/blob/develop/migrations/1717233556_v0.23_migrate.go) would have been already executed._

**Reply by lumosminima** [reply]
Previously, If I needed to delete logs I just deleted the logs.db so maybe doing minimal aux migrations or adding some command to delete app logs would work great in case auxiliary.db would not be ephemeral in future.

**Reply by pnmcosta** [reply]
> Previously, If I needed to delete logs I just deleted the logs.db

Same here, guess we should not delete `auxiliary.db` going forth

> adding some command to delete app logs 

Thinking `./pocketbase logs purge` to remove all logs would be great for me

**Reply by ganigeorgiev** [reply]
General support to reapply specific migrations conditionally was added in the `develop` branch and it will be available with the next `rc3` release sometime later today.

It will remain deliberately undocumented to avoid getting abused by users but if a prominent and general enough use case show up I may reconsider to make it "official".
The reapply works by checking the specific migration condition and deleting the existing applied history entry (it will be added back to the history if the reapplied migration succeed).

The `auxiliary.db` could be considered "ephemeral", aka. users are free to delete it and it will be recreated with the minimal required schema on app startup.

**Reply by lumosminima** [reply]
That's great but I wonder why was existing logs.db functionality deprecated? wouldn't it be better to add the new auxiliary.db in addition to logs.db as previously it was a fool proof way of clearing logs since there was no delete button on admin UI. Also separation of future auxiliary.db concerns.

**Reply by ganigeorgiev** [reply]
Because there are plans to store other system things beside logs and having 3 databases will require duplicating one extra time almost all basic db `core.App` interface methods, like `app.Save()`/`app.AuxSave()`, `app.DB()`/`app.AuxDB()`, app.NonconcurrentDB()/`app.AuxNonconcurrentDB()`, `app.RunInTransaction()`/`app.AuxRunInTransaction()`, etc.
